test/ruby/test_exception.rb

```diff
@@ -149,6 +149,41 @@ def self.meow
     end
   end
 
+  def test_exception_with_defined_in_short_circuit_if_condition
+    bug = '[ruby-core:20501]'
+    conds = [
+      "false && defined?(Some::CONSTANT)",
+      "true || defined?(Some::CONSTANT)",
+      "(false && defined?(Some::CONSTANT))",
+      "(true || defined?(Some::CONSTANT))",
```

Comment:
These lines are same as previous lines, except for parentheses.
Necessary?

Reply:
It exercises a different code path, I put a comment about that in my most recent commit on this PR